wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt

```diff
@@ -130,26 +132,35 @@ fun parseFileDescriptor(fileDescriptor: DescriptorProtos.FileDescriptorProto): P
     packageName = fileDescriptor.`package`,
     types = types,
     services = emptyList(),
-    options = parseOptions(fileDescriptor.options),
+    options = parseOptions(fileDescriptor.options, descs),
     syntax = Syntax.PROTO_3,
   )
 }
 
-fun parseMessage(message: DescriptorProtos.DescriptorProto): MessageElement {
+fun parseMessage(message: DescriptorProtos.DescriptorProto, descs: Plugin.DescriptorSource): MessageElement {
+  val nestedTypes = mutableListOf<TypeElement>()
+  for (nestedType in message.nestedTypeList) {
+    nestedTypes.add(parseMessage(nestedType, descs))
+  }
+  // TODO: enums
+//  for (nestedType in message.enumTypeList) {
+//    nestedTypes.add(parseEnum(nestedType, descs))
+//  }
   return MessageElement(
     location = Location.get(message.name),
     name = message.name,
     documentation = "",
-    options = parseOptions(message.options),
+    options = parseOptions(message.options, descs),
     reserveds = emptyList(),
-    fields = parseFields(message.fieldList),
+    fields = parseFields(message.fieldList, descs),
+    nestedTypes = nestedTypes,
     oneOfs = emptyList(),
     extensions = emptyList(),
```

Comment:
Looks like another thing that wire does not support is extensions that are nested inside of a message. This `extensions` field is for extension _ranges_, not `extend` blocks. Their parser does not support extend blocks inside of messages. I can make an independent PR to add support for that. (Along the same vein as adding support for groups to their schema and marshaling code.)

Reply:
Correction: their parser _does_ support extend blocks inside of messages... but they [store the resulting extensions in global scope](https://github.com/square/wire/blob/4.4.1/wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/internal/parser/ProtoParser.kt#L205-L206), as if they were top-level declarations (WTF!??!). I'm pretty sure that is _totally_ wrong... I'm still going through the code in case there is something subtle in there that somehow makes it work correctly with this wonky model.

Reply:
I added a [story to Linear](https://linear.app/bufbuild/issue/TCN-376).